src/rgw/rgw_sal_motr.cc

```diff
@@ -2457,7 +2459,8 @@ int MotrMultipartUpload::delete_parts(const DoutPrefixProvider *dpp)
   do {
     rc = this->list_parts(dpp, store->ctx(), max_parts, marker, &marker, &truncated);
     if (rc == -ENOENT) {
-      rc = -ERR_NO_SUCH_UPLOAD;
+      truncated = false;
+      rc = 0;
```

Comment:
The rc is masked (reset to 0) here to allow aborting an empty multipart upload session.